source/tutorials/packaging-existing-software.md

```diff
@@ -0,0 +1,870 @@
+
+Nix is automatically working with the `Makefile` that comes with `icat`, which indeed lacks an `install` target. The `README` in the `icat` repository only mentions using `make` to build the tool, presumably leaving installation up to us. We've now discovered one limit to what Nix can do for us automatically: it doesn't read minds. Fortunately, it does still make the fix quite straightforward to implement.
+
+If you haven't read the tutorials on creating [ad-hoc](https://nix.dev/tutorials/first-steps/dev-environment) or [declarative](https://nix.dev/tutorials/first-steps/declarative-and-reproducible-developer-environments) development environments, do that now before proceeding through the rest of this tutorial; dropping into a `nix-shell` is a crucial component in the Nix user toolbox, and indispensible for debugging.
+
+To enter a useful development shell, we'll pass the dependencies from `nativeBuildInputs` and `buildInputs` to `nix-shell -p`. We'll also make sure to include `git`, so we can clone the `icat` GitHub repository:
+
+```console
+$ nix-shell -p pkg-config imlib2 xorg.libX11.dev git
+```
+
+After many lines of output about Nix copying dependencies, we can use the following commands to retrieve and build the `icat` source code:
+
+```console
+$ git clone https://github.com/atextor/icat
+$ cd icat
+$ make
+```
+
+In the current `master` branch of `icat`, a warning is thrown when building:
+
+```console
+[nix-shell:~/icat]$ make
+gcc -c -Wall -pedantic -std=c99 -D_DEFAULT_SOURCE `pkg-config --cflags imlib2` -o icat.o icat.c
+icat.c: In function ‘main’:
+icat.c:319:33: warning: ignoring return value of ‘write’ declared with attribute ‘warn_unused_result’ [-Wunused-result]
+  319 |                                 write(tempfile, &buf, 1);
+      |                                 ^~~~~~~~~~~~~~~~~~~~~~~~
+gcc -o icat icat.o  `pkg-config --libs imlib2`
+```
+
+However, this does not prevent the binary from being produced; an `icat` executable is now present in the local directory, and it's up to us to decide what to do with it.
+
+### installPhase
+In order to make packages available for other packages to depend on, Nix copies everything to the Nix store (at `/nix/store`), and symlinks them from there into build contexts and development environments.
+
+The `Makefile` doesn't provide an installation step, so we must produce one for our derivation, using the [`installPhase` attribute](https://nixos.org/manual/nixpkgs/stable/#ssec-install-phase), which contains a list of command strings to execute to accomplish the installation.
+
+The `icat` executable is only used at runtime, and isn't a compile-time input for anything else at this point, so we only need to concern ourselves with the `bin` output. In Nix, the result of a build is copied to a location stored in the `$out` variable accessible in the derivation's component scripts; we'll create a `bin` directory within that, and then copy our `icat` executable there:
```

Comment:
Don't hint at things that are not relevant yet. Just say we have to copy the build results to the output path and show how to do it.